packages/next/pages/_document.tsx

```diff
@@ -176,8 +176,36 @@ export class Head extends Component<OriginProps> {
     const { page, buildId, dynamicBuildId } = __NEXT_DATA__
     const isDirtyAmp = amphtml && !__NEXT_DATA__.query.amp
 
-    let { head } = this.context._documentProps
+    let head: any = []
+    if (this.context._documentProps.head) {
+      head  = this.context._documentProps.head
+    }
     let children = this.props.children
+
+    children = React.Children.toArray(children).filter((child: any) => {
```

Comment:
Instead of looping over `head` fo each `child` you could move this below the loop we already do over `head` and create a map of the tags in `head` and check against the created map